Features/NotesFeature/NotesViewModel.swift

```diff
@@ -66,10 +71,40 @@ final class NotesViewModel: ObservableObject {
         }
     }
 
+    func prepareNotesForSharing() async -> String {
+        do {
+            return try await crasher.recordError("Failed to share notes") {
+                var notesText = ""
+                for note in await notes {
+                    let noteText = if let noteContent = note.note.note, noteContent != "" {
+                        "Note: \(noteContent.trimmingCharacters(in: .newlines))\n"
+                    } else {
+                        ""
+                    }
+                    
+                    let verses = try await textRetriever.textForVerses(Array(note.note.verses))
+                        .filter { !$0.isEmpty }
+                        .map { $0.hasSuffix("﴾") ? String($0.dropLast()) : $0 }
+                        .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+                                    
+                                    
+                    let verseText = verses.joined(separator: "\n")
+                    
+                    notesText += "\(noteText)\(verseText)\n\n"
+                }
+                
+                return notesText
+            }
+        } catch {
```

Comment:
I would rather not catch the error and show an error alert to the user. Use: https://github.com/quran/quran-ios/blob/4df3eb4a8c91571c18d9e8e38592bf7a0bad6b50/UI/NoorUI/BaseControllers/UIViewController%2BError.swift#L26